mmhuman3d/data/data_structures/smc_reader.py

```diff
@@ -519,3 +533,153 @@ def get_depth_floor(self, device_id: int) -> dict:
+        """Get keypoints3d computed by smc processing pipeline.
+
+        Args:
+            frame_id (int, list or None, optional):
+                int: frame id of one selected frame
+                list: a list of frame id
+                None: all frames will be returned
+                Defaults to None.
+
+        Returns:
+            Tuple[np.ndarray, np.ndarray]: keypoints3d and its mask
+        """
+        kps3d_dict = self.smc['Keypoints3D']
+        keypoints3d = kps3d_dict['keypoints3d'][...]
+        keypoints3d_mask = kps3d_dict['keypoints3d_mask'][...]
+
+        if frame_id is not None:
+            if isinstance(frame_id, int):
+                frame_id = [frame_id]
+            keypoints3d = keypoints3d[frame_id, ...]
+        return keypoints3d, keypoints3d_mask
+
+    def get_smpl_num(self):
+        return self.smpl_num
+
+    def get_smpl_created_time(self):
+        return self.smpl_created_time
+
+    def get_smpl(self, frame_id=None):
+        """Get SMPL computed by smc processing pipeline.
+
+        Args:
+            frame_id (int, list or None, optional):
+                int: frame id of one selected frame
+                list: a list of frame id
+                None: all frames will be returned
+                Defaults to None.
+
+        Returns:
+            dict:
+                A dict with 'global_orient', 'body_pose', 'transl' and 'betas'
+        """
+        smpl_dict = self.smc['SMPL']
+        global_orient = smpl_dict['global_orient'][...]
+        body_pose = smpl_dict['body_pose'][...]
+        transl = smpl_dict['transl'][...]
+        betas = smpl_dict['betas'][...]
+        if frame_id is not None:
+            if isinstance(frame_id, int):
+                frame_id = [frame_id]
+            body_pose = body_pose[frame_id, ...]
+            global_orient = global_orient[frame_id, ...]
+            transl = transl[frame_id, ...]
+        smpl_dict = dict(
+            global_orient=global_orient,
+            body_pose=body_pose,
+            transl=transl,
+            betas=betas)
+        return smpl_dict
+
+    def get_human_data(self, device=None, device_id=None, frame_id=None):
+        """Get keypoints2d projected from keypoints3d.
+
+        Args:
+            device (str):
+                Device name, should be Kinect or iPhone.
+            device_id (int):
+                ID of a device, starts from 0.
+            frame_id (int, list or None, optional):
+                int: frame id of one selected frame
+                list: a list of frame id
+                None: all frames will be returned
+                Defaults to None.
+
+        Returns:
+            HumanData: all annotated information packed in one human_data
+        """
+        human_data = HumanData()
+        # get keypoints3d
+        convention = self.get_keypoints_convention()
+        keypoints3d, keypoints3d_mask = self.get_keypoints3d(frame_id)
+        keypoints3d, keypoints3d_mask = convert_kps(
```

Comment:
`keypoints3d` in `human_data` should be in the device's coordinate system. But this can be fixed in my PR. We can leave it as it is now.

Reply:
OK. It is now in world coordinate.